src/nmcli/connections.c

```diff
@@ -5236,9 +5244,70 @@ nmc_process_connection_properties(NmCli              *nmc,
                                   gboolean            allow_setting_removal,
                                   GError            **error)
 {
+    const char *const   *args;
+    gs_free const char **to_free = NULL;
+
     /* First check if we have a port-type, as this would mean we will not
      * have ip properties but possibly others, port-type specific.
+     * Then check connection.type, as port-type might be deduced from it.
      */
+    if (!nmc->complete) {
+        const char       **dst;
+        const char *const *src;
+        const char        *option;
+
+        to_free = g_malloc(*argc * sizeof(const char *) + 1);
+        args = dst = to_free;
+
+        for (src = *argv; src < *argv + *argc - 1; src += 2) {
+            option = (**src == '+' || **src == '-') ? *src + 1 : *src;
+            if (NM_IN_STRSET(option,
+                             "connection.port-type",
```

Comment:
Consider using macros for setting and property names (e.g. `NM_SETTING_CONNECTION_SETTING_NAME "." NM_SETTING_CONNECTION_PORT_TYPE` . It is less readable, but allows one to grep the code for the property or the setting.
Apart from this optional suggestion, the rest of the branch LGTM.